crate/kmip/src/crypto/elliptic_curves/kmip_requests.rs

```diff
@@ -10,24 +14,77 @@ use crate::{
     },
 };
 
+// `unused_variables` is on because when FIPS features is disabled, `curve` is
+// unused.
+#[allow(unused_variables)]
+/// Builds correct usage mask depending on the curve. In FIPS mode, curves are
+/// restricted to certain usage.
+///
+/// For more information see documents
+/// - NIST.SP.800-186 - Section 3.1.2 table 2.
+/// - NIST.FIPS.186-5
+fn build_mask_from_curve(curve: RecommendedCurve) -> Result<CryptographicUsageMask, KmipError> {
+    #[cfg(feature = "fips")]
+    let mask = match curve {
+        RecommendedCurve::P192
+        | RecommendedCurve::P224
+        | RecommendedCurve::P256
+        | RecommendedCurve::P384
+        | RecommendedCurve::P521 => FIPS_ECC_USAGE_MASK_SIGN_AND_DH,
+        RecommendedCurve::CURVEED25519 | RecommendedCurve::CURVEED448 => FIPS_ECC_USAGE_MASK_SIGN,
+        other => kmip_bail!(
+            "Building mask from unsupported curve in FIPS mode: curve {}",
+            other
+        ),
+    };
+    #[cfg(not(feature = "fips"))]
+    let mask = CryptographicUsageMask::Unrestricted;
+
+    Ok(mask)
+}
+
+/// Builds correct algorithm depending on the curve. In FIPS mode, curves are
+/// restricted to certain algorithms.
+///
+/// For more information see documents
+/// - NIST.SP.800-186 - Section 3.1.2 table 2.
+/// - NIST.FIPS.186-5
+///
+/// TODO - Discrimine between EC, ECDH and ECDSA.
+fn build_algorithm_from_curve(
+    curve: RecommendedCurve,
+) -> Result<CryptographicAlgorithm, KmipError> {
+    let algorithm = match curve {
+        RecommendedCurve::P192
+        | RecommendedCurve::P224
+        | RecommendedCurve::P256
+        | RecommendedCurve::P384
+        | RecommendedCurve::P521
+        | RecommendedCurve::CURVE25519
+        | RecommendedCurve::CURVE448 => CryptographicAlgorithm::EC,
+        RecommendedCurve::CURVEED25519 => CryptographicAlgorithm::Ed25519,
+        RecommendedCurve::CURVEED448 => CryptographicAlgorithm::Ed448,
+        _ => CryptographicAlgorithm::EC,
+    };
+
+    Ok(algorithm)
+}
```

Comment:
Mmmmh I don't know, this part of the code is really sensitive to changes since for the moment, the "default" ECC algorithm is `EC` which is not flexible.

I don't know where from should the information on whether this should be `ECDH`, `ECDSA` or `EC` for P-curves so I defaulted to EC.